src/datamodel/serializer/UUri.cpp

```diff
@@ -8,5 +8,97 @@
 // https://www.apache.org/licenses/LICENSE-2.0
 //
 // SPDX-License-Identifier: Apache-2.0
-
 #include "up-cpp/datamodel/serializer/UUri.h"
+
+#include <sstream>
+#include <vector>
+
+#include "up-cpp/datamodel/validator/UUri.h"
+
+std::string uprotocol::datamodel::serializer::uri::AsString::serialize(
+    const uprotocol::v1::UUri& uri) {
+	std::string uriAsString = "";
+
+	// Following code is disabled because validator for UUri is not yet
+	// implemented.
+	// TODO: Enable following code once validator for UUri is implemented.
+	// auto [result, maybe_reason] =
+	// uprotocol::datamodel::validator::uri::isEmpty(uri); if(!result) {
+	//     return uriAsString;
+	// }
+
+	std::stringstream ss;
+	ss << std::hex << std::uppercase << uri.ue_id();
+	const std::string ue_id = ss.str();
+	ss.str("");
+
+	ss << std::hex << std::uppercase << uri.ue_version_major();
+	const std::string ue_version_major = ss.str();
+	ss.str("");
+
+	ss << std::hex << std::uppercase << uri.resource_id();
+	const std::string resource_id = ss.str();
+	ss.str("");
+
+	// TODO: Use uprotocol::datamodel::validator::uri::isLocal(uri) instead of
+	// uri.authority_name().empty()
+	if (uri.authority_name().empty()) {
+		uriAsString = "/" + ue_id + "/" + ue_version_major + "/" + resource_id;
+	} else {
+		uriAsString = "//" + uri.authority_name() + "/" + ue_id + "/" +
+		              ue_version_major + "/" + resource_id;
+	}
+	return uriAsString;
+}
+
+uprotocol::v1::UUri
+uprotocol::datamodel::serializer::uri::AsString::deserialize(
+    const std::string& uriAsString) {
+	uprotocol::v1::UUri uri;
+	if (uriAsString.empty()) {
+		return uri;
+	}
+
+	std::string uriAsStr = uriAsString;
+	const size_t pos = uriAsString.find(':');
+
+	if (pos != std::string::npos) {
+		uriAsStr = uriAsStr.substr(pos + 1);
+	}
+
+	std::istringstream ss(uriAsStr);
+	std::string token;
+	std::vector<std::string> tokens;
+
+	while (std::getline(ss, token, '/')) {
+		if (!token.empty()) {
+			tokens.push_back(token);
```

Comment:
You are right. As per your suggestion I have modified original function to use std::string_view to avoid copy. But by code readability point of view , it looks complex compare to the original one which does not use string_view. 
I have commented out the original one and also uploaded new one with std::string_view approach. Let me know what you think, which one looks more optimized and based on your suggestion I will keep one.

Reply:
Looking at the string_view implementation, it is still copying strings into a vector of tokens. I also don't find either implementation particularly readable, although that's more a problem of lackluster string parsing utilities in C++.

I made [an attempt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxCAAbACspAAOqAqETgwe3r56KWmOAkEh4SxRMQm2mPYFDEIETMQEWT5%2BXJXVGXUNBEVhkdFxiQr1jc05bcPdvSVlgwCUtqhexMjsHOYAzMHI3lgA1CYbbsPEwcCH2CYaAIKb27uYB0cnZwD6AG54mADuF1e3Zi2DB2Xn2hzcyGG%2BEEfxud2BDyebnemAcJFhAKBILBzwIWFUqySBAx8Oxj3ByAQDTQDHeJJuzDYCiSTFWe3eXAOAHYrDcTl4HHs5HJTtzedc9pK9lCQCAXkY9kwvAQECRCABPQ7iqV7LzBAgbMyvAi6zCvPDoLX/HV6wSG42mj7RNICK03G36%2B0m4iYBRLFZmi1u25cgAiwZMYf%2B/w5suFoqwSlOYjwAC9MBAacNpXjZfLzmZYrrTtcFHVTkY5mLoxoAJx4Kh7CDLPCl8tnAB0mBYRPVEDmVcj2p1UpVxFQ3xz6FlwXeKfQrwawB8jAIEHMZjchgYqBNieieBT6b23d7OYrBbMcwjdcj4bh7qlWYImFUSWIU7zBAvHy%2Bk4UlLdkwrzvpgNCqE8oYHGYZheEkIAAPQIRuwY6s%2Br7vp%2Bcrfm8nw/HsPosLuZqgeBkHQWYSEoRsw6Suhb4fjK2E/nh/6YMAbCCK8SgssQTBEB%2BhxQRuyEwTeEpSkhewAPJJNE/FnHsAiKueimsXsRCKu8qAWnsaBJF8CiKgw6AEZgYi0Kgog1Ao1pSkx%2Ba/vhLYAGp/s2JZljhlbiTqUluD6/GPEwqkKupVDjiwGkII8kyNEpjauX%2BGmoNFjwhKoJo8Q0/EkB2dmSlJ1zoHOwK%2BsWeBufhmkqo8eTpMpTBUC%2BH61dKmC8blxD5Y%2BkpKppr7fqywjsZxJpCQc8RWIWSU/KQ7Ucau3EdTlAkmPEob9nsAC0FxYY56lDgVI7Poqyqpa45Gzd8HY0CZEBKItXHZXxAnXjRx06g2TaXUJE3XR2DB5IOPKfSOkpjhOWGzvOi7EMuY3rjBHheLQpk7iag18YKj1jXsEWoFFYDrGJvXg%2BT1j7d5wAeZVf6DpYFHE3s3yEAg7WdQJezEyhZPkyOlMOdTD2jUtL1dQzFhMyTV6oRTUZ81K/Wpbjq5XacVU3QoXgRCcEAaPNrjvbRI7XerdM/B22u69%2BECuNYXDG2Dko%2BgQywMAtY1y5Kd7ewcD61rGIDxngFW%2BZJCF7C5B5UOqObdMZplY8NezXMqqqnAQ6rzd9oFKDCZNSQA6mzurwVJAExSwTDHd9tOa1bOt6wb0qAdXIE%2BuBVtphmA6QRNldAR3YF4KoIMm35kfFaVbIstmbWkaPzsVZr5sN9beuD%2B3i%2BqN36b9k7iuSi2Hb9RnGrkcnDhCKLgj9uHhWR6zKp7FJhHEXsO/HXeJ60Eoex1wBhvW2Ld34vmHl3NI%2B8%2B5/Q2FBMBJFO6j3HsvIqJVDCzyYPPGKn8kGqGXmbf6Gs/yNxtsQCACCIGjz3r3Q%2BElwYnzPmqLOl9MrYxGk9NcdDjpSWfuzSOllRC0FwSPfBZMf5VH/oA4hltgHkJbqrZ6K1Xp5SgbQ/ucDPZi2URLasR9X5T3QWVT%2BWCTQLzwQQmRk4iEWy1k3W2iiCDLU5qonuB8/Y8MjgAWTwAoNICofQAEc9Q%2BlMl/cRYZf7/yOvoyG/5cwgBhrQC0cMEariRmYUMukMb42CEnN8qIXymTkAAJQAJLx3isEUKoBuYy2XjqQWCT8z13ptBKWG4eYwToY0hWtxC6RwAGIkBUsAPAKIPaOPmjSFE8U2pzm8I8TShgAGCHYtEROOYSCPEIKsvYDAfDHWVlowQAAVVAMwNkTXWlYDamYBDZiFixZKCh5rK3MEWA5LAUH6OOTc1whtTAbXIkxAmLBXiUgaAoB6HZ0D8SYP2eaChYXwvvpYZFajEX7J8PNLgsQenk1OhEVAng9ICFmU41AABrcidtkAaKgkxaIKwjphgZrED5P0TIMulCi%2BoaKLAYrcQOP2X1GwQDAGAGZ0RKVUp%2BfQ/m0VxzxOnIk2ksMlwrjvhuQZTA6CYFMppaV8VHEpWxaUAYFEGkU0Zk8s4D0RUfX0b7CJ4YnX9IVVJbAbCU6GFMsasxOCfTZlQI2NqpSKn5mOo485lzyFXw4YjBYFUOxeEDOgAlkoY0XP6PGn119b5cPmifNNTpiAugYJmgxexTlBt9P6Nkvi0p7AMI8wtmz0CoHKjkqkKIVJDToIpZCZNs1xtaXNFNwaG3pu4QHV27sw7updR6msK6HweqkkIZQoYAAa21Bn6o8EkdUpxgAIAIKcwaIA9hmA0GYSQewPCCFOBEc65azVtWwDsPASR/7DK8CZBSrobhURA5HG%2BSycGhGkqcspbhsC5PoA6vY%2BAXivqKSzUuKom3fBIDS/gH4mDoHwDUMQ/wpLBAI9XGoZlgANGhMAMlx7T3nqUt8EI5aEA/p6tcUDvHI61qbe%2BVAwA%2BJRT9c21kaAeyGHVIpajB4xBGQaI8au%2BwmBzjoEwCI9BdQmQ2bVcjkcWosCMqG5t1wWSAT2EEVYDB/7R3LRkG9HYNAsy45SABynNMGB08FYkYG9jnoIL%2BxCCFvgRdPlZmKHYSDAAQskuzSgFAIQCHB7AoQhDYG2mYVzRmjN7C3bu7atnGBKG2mUrAggGxfBiKnaLmAct5fXaSRE4JgAvmGAhDrvoCAdgQPSa4jJfRWfJKDddOwsFGUvcMEON9kzJPTB%2Ba9SQdaJY0r1s4soZvjXG5Z8cL4HAGpADWWsUkSkAaUsYuSH5TnYCEKc14gyeNXDO5HRQBqzVKBNHBPS9AVmME%2BOOBgY0jI3Y28MF7dZtK6RvvIJIW1UCzNOGCUGfTXsw9MpehooYJwMER8ji0Y2rDo4Dudy7AhZ4bNfKiZUznzNtU6yaJgSQkiJaAwwKHb2hRKFMgR9qbskgJUVLQYRTPkV7AABJYMVMgVYfiP1caMsESYZUuc7bm4pxb0QtoTSwFQJUtBiRLrrAAPw1yKPA83Dza/IYOTR%2BvDfG/FKd8nHtKd1Wp6oWnNGGc4KZ4qVn7Oahc6kh90yAhaBx358ASyEQxAC9%2B0sAgaR9gM968i07kxHD0sx4VzA8OdtCD1C%2BXXaP7z9NrNnvAuedJY/MsQXH7Gi8l97tWdHUYTfXDuw9p7EALenGt0eaI81QyYCH9rub1MS4qhEFXTA8rXunTtQqLwLY2zU3IhuOCYWuC1ly3igAHB2LgJ%2BNAIS4FwDQ5/BmDIQof6/GhqIu7rMv5pm%2BMKFINWnFUzC47XJgj3wP1iGP1Pyv2f1O1OltANCNExgKSO3QDkGwDKWEk0Q0FUEv2vxvFrCgM9FgJPHgKKS8SYG0BIEcwrXInQNv2wNwLtHwM/wQJKXrWWFWBQMoNUAfywK7yXweWZ3OgqnIn3AWzcTX08nbB8hN1rGuCECyxKUe2wAAEU7ZCDv904/9i1ThT41DM51RZ0pCZDsA5DXhFDlC5IECkCUCNC8BU0Z1sDpDZD5ClCGCiCSCyDnQMgrCbCy0K09D7DDDHDTCv90AmC/QWDMBLDJ1mCAxzQM0IxSdu97tHtBl%2B9etNdhCltR9x8tce4p8LwZ9VRlQ59uwF89FK839VV8xdR18vILwt8YIkIgCT8QCT8z8L8r8b879OCn8xJJDyivxajnDVDf8dC6izBGij8WjwCeiX8cDeDdQ8CHRBjEDkDUCoJ0DMCn9ei5joCvQCCzCXDSDiByDnMJoqDBkaDtiFi4D9iDUQjp02DTiODH9xIeD7M%2BDNIWxBDfRsj95RDWwaizhfCDCjCTClif9z4s5PCmEdCgSHDjCnCVDliIiS1bDJC/CQSESbj0BiDDjjiBBPDS1ZkfC7DgSAili7iwjkTNCp0wiYjuF4ie8kiUjZtLcJ8e4yA9gx82T0xsAews5cjATSjXjHl39ai/iN9aiACzAICA4%2BjmJFIwTtCL4pSZSyjLi6DFjESLDVi9h0DNiZjaCYDNSsScS3CnNlJHj9TID1SjTrigiKSAwHi0DVArSWs6x0THtTlJcSlpIi4IAhCbcRDqjxDgBk0mIkkUlNUxp6SK9/hGS%2B8B8rcfiR9OSsj0jMABSjB8jQhUBwS/9F9X85iV9GNxSASFQpS2jr8EJb979njpjrS3iqYBjES8yRiVT6zZSbTdiljtT2CNiLjGydj6DETTSjj3CLTnTqCtjByri9j7SojWCdT0CuiXjCzGzjlPi9dvj0zmxgzqZYT/D4TAiEDWyNQoSlSs4DyMTjyiltSCTUSZiPSjyljRy8SGB7zvCMgryyTESHTFzPCaToiLQYzV14zkjEzuSUyuTkzMzgB8iyl1Vkl0A41pIqBrh4YtUCACzudlBxxPgsBTJ4hFQMLQdVlhhzII9GwH1vgYoPZoSL5XxfEAtK9iyqixDN8KyL998mjQDWiNjqy78ayVyOy119C4SvSfS/SAzh9dz2KLwwyEkIyFwozVxZ1K8w8EM2Cm0WBfF/FgAOw9hcKkcidTINhiK0lBBlc3iKLhdqLaKzphiGLVAmKudSyQzRiGjuKJiwCOjaysCRLXsnyJLfT/TtzAzfi9z5KkVFLELIySLVLVycK8KTK9gH0VLLKyKXwiNhczKaLGAHKIS45O1u1dwCCXLTs3KOLNERJ%2BKayhK6zpS0TSTXhgqpKwqZLKqoroZYrlL4rBA1LXsNK9gtKjIdK/FOxDLkqCKb1zLMKrLyLsrzNcr7L6KWFGLIcKrIrFIKz6r/LGrHzmrWrQqkxwqMxOrATorVUlLUlMKQL10wLmT5BWTkyOToL0zYL4Ker0KLKsKhS6wpJccSqsoRgTQ%2BFjI45h0WLRTFJzryzqrACvLmifKBKUbhL9rTsgrvSQrpLtdZL/iQyFKrqeqbrozErN0QajIwaiBUpq4GAIbNq5Ltr4bKJPLgDeKr8Ubdruj0aA5MbJLjrky8aJSLruqFk4qfqBr/r3tNKoJhN8LPsm0lKDKyksoCi0Y9goggtXxzUDxTAA5YbGNOLxikbWimckDUaGqSTxKsa2qTqOqtrKxLqZxib0qsKybvFXCPw3zcFjLprFaerlbValh1bNaYoIIvldaGb8aqrhJ6iuK2bJjz9%2BL1pLgzB4g/LuarbDyjqcagzGbHbRaNU%2Bq3buCpa9g/zHg2C5aUr/axb0BA7pQ1bTJQ7taI7Tg9bK8DaPL46eLE7KzfLVB1RUxVTArDqbaBadyDbCbna66SaEql14if5q1htmRWRMB/gOAFhaBOB4heA/BuBeBUBOA3BrBBZp1oINgeBSACBNBN6FgqUQB4gDZt6OBJA97b7SAj6OBeAFAQADYb6OAtAFg4BYAYBEAUBCYDJ6AyAKB7kex9UYhgAuBAQ%2BA6AWpf6IAIgP6IhggGh1ROAr6cHmBiB1RpIIhtBCkCHeBpMxppIGAo8P6sBX1gAtxRdf6D7SAsBabY81gtBOG8AfQHBxlfQP6adkBlReHeB9QqgP7kkIg%2BISGPAsAP6cIWAqHSBZliUlAx9uHkkjBb6FgqADBgAFBNZZJGB1H%2BBBARAxB2ApAZBBAPs1AP7dA2gDB9HTBLBrB9A8BdZ4AIBhsQBgASUeGmcNGBhvwAM5UFhUAiQMh2HtptooQhJPHpoLAuAuQdpaAethhXhq5qltppJgBD7CdEx4AFg7BCkMgXATIxhWhSBAhgg%2BgLUYg2h6oag6nchUgGoGBphc09BKmhGBAuhRhPAWgBmqgqnhmQa%2BmWmBmQbOmJgZmmm40JAKnp01n9Ad737AHD7OA9gODYhtpYgH1gA5c9hkGOwzKIBcBCARlNhHZeAAGgHSAIAkBWdyBKBJsjBtoqN%2BJ1GDU7mUc9AaAjdohtoYoiNnQP6rHhAhE7HpBYWnH1BdnSBXHr6fRMALQ/6t7tnSB96%2BGv7cKOpxx5c/QPxpJlRVsTRDnjnTnznLnrnbmuYHn5oPB4HoGL7Hnr6DH77H7n7OA37SA1Gn78WP6v6f6/6eXdngGwG3mQBk9qXPm4GoHohQhWA1haWTm9gzn6UpAT9eBAWBJsW2hYWbHxB7GkWlBnHUX0Xvg%2BJ4ID7cWOBd6xXUWv7KWQtlRhctX6W9Xctrn2XVXBJARuXnm77SBIWsAYhEUX6hWRWDYCW9nv7bApXw2BxSAH7RWX6NgdnCXOAnneWtmOAzA83k3C2ZWFgiTqnJAgA%3D%3D%3D) at modifying the string_view version of the deserializer to a) remove the extra copies and b) reorder the steps to make the process more readable. I'm not sure how much more readable it is, but it definitely avoids the copies. What do you think?

Reply:
@pranavishere2 - after our conversations, I had expected that you were going to take the code from the link provided and, with few other changes, move the lambdas out to be functions in the anonymous namespace. Can you clarify why such significant changes were needed?

For reference, I though we had agreed on something that [looked like this](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxCCSAGykAA6oCoRODB7evnrJqY4CQSHhLFEx8baY9vkMQgRMxASZPn5cFVXptfUEhWGR0bEJCnUNTdmtQ109xaUDAJS2qF7EyOwc5gDMwcjeWADUJutuQ8TBwAfYJhoAghtbO5j7h8enAPoAbniYAO7nlzdmmwY2y8ewObmQQ3wgl%2B11uQPujzcb0wDhIMP%2BgOBoKeBCwqhWiQI6LhWIeYOQCHqaAYb2J12YbAUiSYK12by4%2BwA7FZrscvA5dnI5CcuTyrrsJbtISAQM8jLsmF4CAgSIQAJ4HMWS3ZeYIEdZmF4EHWYF54dCav7a3WCA1Gk3vaKpASW67WvV243ETAKRbLU3m103TkAESDJlDfz%2BDJ9zNZEa1kupQ0wqkSxCluJlcuA70%2BXylFMwLCYL3TmBoqkeIf2ZjMXkSIAA9E3zGYg9rkwRU%2BnM%2BhswQTkY899dt6WKhu2XvZXq7WzC22x2kwIU2mM9LZYPXh9R0pgGxBC8lMziEwiBmDjW2626%2BG3ZKW7s3N7zw8mJmh8A2fndlRiKgLC7MqDwTA0uyoFQOonAAar%2BRDAQgDwhKoxonvU54kAAdFaj5NrsVzoG8hgrAo0F4HBo4ISBuy5GkAgKlQ3YZjR6FnheOEPhKm45iOBapoOLLCJgB6MAQEA8duw67j8ZhxORfGzKKuHahKXYKkqqC7K4c5LHgfFYTQDDoBJImHgQx6YKemHELMy6qbseBQRAOlXleCkyVhDC5EpCYqQ5krKgBBabsExG0OaLz1MAPhiRAbYeF4tDoLsDCTtpqFngK%2B7mX%2BAFAWAax3lxAUOdYfYDl%2BEB6YptYWPOhW7F8hAIFKVkYReuyFUuJWldq5WSVVOViZZ1kXr5lgNUVZh2esiZlZGvWSoqCHDYIuknAZCheBExwQBopDacZs3zapNUyRt%2Bmedtu2Di5xnWFwJ3%2Bdq3oEEsDBtaJ0Jzf5EZhr9sIlU%2BABiJAKrswB4Min1rQQh3Usi4E0eFXgPAhhiOYIInRAqxmZiQDyEFjqU%2BP5byoOaX3mQAKqg0zRBA6mDTuv4KIdNr6oaRJyaTLC%2BdyL3LZp%2BwAKwWK4h0oiYovXusNabv%2BgEvBS9QKBJWHoOeTAQPMUqa9ruvWAoWGpAAXpguuHQwPiHVwcTPUtamrsaESoJ4uyI9EFmoAA1nOLnINW7mbtEyx%2BaGE1xOY8n3Slbny/rWt1Eblgm%2BbluzI74oBU5uwQGAYBew0Lx%2BwLp2lUFqAhVmIBhWIkXRbFgjxXWINMHQmApQhxdoWZYnAVpNslP085C31dUVVuQ1Z/Zqn/X9kaA8GYZA1c7IykKIpYEoJwNxbTMu1POYxwpTAKMeUnAOXUYaAAnHn1Wbefl9flhxaEmqus307iHBVP9cIroCisQGK5lW5mDcIYNKxod7RDwPvB4H8CBqk/KcNs2dLh3wXmvbUT4ADyiRojnlOBBT6H4cw/iolpJgFMqZoESJ8MihgUqvloLQVAohqgKH8izaSv5zr5ifvpF%2BOZMH3w3iALeeByL3hzhKJ8MF4FUFQWBY0LCMqCQFFcJUKoTgoMOnncsShoTA3wgAdRajqRsT4FBFhLP5R%2Bgjvimx2ntA6hYkIlmnBWPAqhTZ4APlnYOic7FeNLOWSsP95GqSfIRYiQIHjMiGIhJJM4/Hjw8r%2BdyziviuNusQCS9iInpP8Rnb%2Bc9rQnCwitPR6o5wCSysJb64lcniJiU%2BZqypdhPnHJONJvjVCLxrJUJQjlnK5Pye4w6fSpyRL8QEoJvl5buVmaaeZqhomlTiUREiSTz7GhohszJuTLpbTcXdNZPjKyLMztnUqekam6NVCghpmUhJCH7i3Npci8H4S6a1fCHDRC0FoqU4Z2laBjKcZta6FzCkeLhqNDq2FynBITgrL5Fk2I2S2QFHZCTWTJMOUhMFgyTmwuyYnSZN09pIpxRxNF9zJT%2BSfAAWTwAoVI8pvQAEddTehSsckq/1IVjL8r/KuNd%2Bx1xpA3YBTdwFthDFTGBf5ggpR7CibsKU5AACUACSmYugkxzCALq01Mn9Umnw4AwjarlTbN1OszKJQ4JuGY3Y2B3kCg0b3VJY4fTGkggG/VRqT4lThnTBmhTGkfKxfamSetHlozNOgV11MxLRr6LGn1zTwFtI5tU1NSNnQMAzU%2BGmpLvS%2BiWKyTlAaDApLhnjFK6BUA%2BlSulSkyIIaCToKQ1skasXZpHoU3JRa8BYRrX6FYab2lYLeh9WRy93W3w9Ru9dfwnxCGUCGAAGgAWhBp3DwiQ1QnGAAgAgNMBLmrMBoMwkhnyyBOBETSxAyLUVJdgbYeBEhjLBl4YyJCXTXEXOB/Cnz0aktCPgmmBq3DYHVfQCSSl8DPHfTqpqVjlQNq%2BCQf2/AMxMHQPgaoYht34WCMRks1RA3AHqFCb8DCL14CvcGr4IRP0IH/ZxK4EGBP4SrQ29MqBgBniAhomiLI0AsGZAwNUpC6PwLEMw70uwSx7FoR3AwER6A6mMrjECVHgLRBYGRENNErhxlJUEFYDAxlKM/ekXYZgsIaCarxikjlmHEToEwfT74iSQd2NeggAHmxNi%2BDFmptnMBYRIMAJsEUHNKAUE2AIiHsChCENgQ97mNBUdM7ug9h77OMCUIeg1WBBBOU%2BDEAi8WCsea3QCO4IIySHGAN2IYTYetBqwggOkVwYxMhZGSQWa9tjnzIreoY0jPl7wihbDM5rEg7VS2ZoYpwZTzeNBKmzAFuwOC7iAW%2Bd8nx6uA2QoluMabYCEDTF4IN%2BNYKfIoLug82rGgbJ7egmNGAfAAgwcyZEiEsUGxduhKVPnyESLrCCSMTigkFotD1d8Ye7FvfUEM1cGCI9QMj80k2rDo4u1dm7Ag7sZlTCiJUrmrOkt6%2BoxIiRUugYYG9%2B%2BH2lApWIz9v7IaxCgpZybXYAAJc%2BCpkCkS/VpPDZFggTESdzu%2B%2B3FuqZW4zZZNYsBUEVLQIkq774AD8NfCjwEthB2vCm692Prw3xuxQU/wtdz61Okm4zp8gBnDEmfoyDQqNnHPqhq9519gQtBUEC%2BABwiIYhBeJAgkqVIewrNBpNhdiYjgg5Y7h3IRI%2B2hC6m7IjiOq8Mc57wHnymKUcfEDx1x4vpfM7KXJ0vF31wHtPZexAC3JxreILILsEMmAh/a8W1fSxyoRBeMwNErBzNa6UK8HpK4ChahfjnG2BsUWuB33c/bAAHFhLgZ%2BNBNi4FwDQl%2BQYgybMf2/RW7wm7vsvmVlCtWnfQDo5ULzUF3I2wD8j84hT9z8b8lw391JOZPQMoiEf85BsADU5YawNBVBr9b97x38j5YDuZ4DtUu42UmBtASBnMy05x0D79sCYCPR8Dv8dU9UfRZ1MAUDKDVAn8sCTcMd1IVotI9I5w4FltAlLY18TgN8t9ThxE74rghA8s9VntsAABFFyNMQg3/Z5fRNUSdJ5f/LQ6Q2Q%2BQxQlQhgruJAlAnQ1Nc0AwuQ7ABQl4ZQ1QhAnVYg0g4gcg9ISw00UtdIGwowhwkwtQn/Jg2tf0Cw8iadZgutAMdNcMcnbvR7Z7EGfvINTXYQ1bQ6MfCfEQqfL8GfFUJUOfYsBfZSIGHAxzNCFfK%2BHUdfTfaooAusFsEAs/MAs/C/K/G/O/B/Tgl/dsaAo%2BW1Agn/P/OpV5BoswZok/NoyA1/LvcolJPA%2B0Uw9Acw1A3YdAzAorfoionUOgpYoIlwkgsgp0VzdyKgkGGg3AvY40ZYkIlgtgs4jg5/ORJfI%2BPg8iQQn0LXEQ6qWoyQowPwuw4wpw9QkYgAnQ2pAAwE%2Bwxw5Y1Yrw%2BdbAwwoEgIkEn/Vw44lzAQBEnwgQaE4E24qIsIkMHQmdaIxE1deIq4HvJIlIhbS3bIjI0fcfb4i2bAeTFBXIqQ0onggYqo7fMQvACQ%2BoxOJcWYi7D/SqUhZYsErQnfO8cUso2g20egg4sw5AtY9ArYuY5UrmfY5wogo49wk4hiR47UiUq4lU/U9Qu46Ih4xOLUl4%2B%2BZE%2BwmmCXPVfBcxCAIQm3H4wU4Ur8PWUKOVIBEBMBMSTBKkmkvvAfK3Vk6ITIlk9IzALkowfI0IVAWU9URfe%2BSU6eUhf0uo7fcYjo2/Jse/R/Z4xU3knYwYmUzQ%2BpcYqAnUy0vUm4tUlYjU9gzYy4nYxY9sg09ADE40rEz6R46g7YhY64oYxgoklYe0tAp4rgrvGslJd4gQ9yH0xBX48QosqQpE2wmEwIwcrMgxCIyE/Qg8/w2Ejs%2BEiIqw2It/F0gkjs4cjw7E%2B87wk08tK8lEm8wc204k0kucmIyMyvP4aM5I2MxkhM5kmC1M4AfIg1EM80GNfBKgK4UBZuAgHMy7fCZQACD4LAFKUWBULCsHLGFMUjCCKCZ9L4JCchBs15VMTlELKvfkgsv4kU68Roq/Q/Fo8A9ozY8sh/Csno5srdZ0w857N0j0r0rc7XHcoUvcgEw6YM8KRuciiMp0vCwUZDNghtFgTlblYALCXYAionEnFKdYMi8MwQJXCozAaikNOihijSPQ%2BpFioYNXQs/478Esvi0AwSm/ESysrA6srBZ8l4WSz070r45MpSgMqQtS2uQBTSuynCnSp8Cyoir7Z9RVMSByqilKENGy%2Bixgdy0Y1BdtTtNVLyggHyri4s0Uxo4SissSqsvouYqKmK%2BS%2BK30g%2BXyq%2BIM1KlChVLSwQBdHnfCJA3YAysiIyrlU4MynKqytzWy7Coq7sZyqCMqtyi8zy1QVixq3cvy%2BUhcDq8Krqi7Hq902KhSv0pq5KgBMasM7CsC9dSCuk%2BQBk%2BMkfLI%2BMhCpCsazCjK3Cp8PHWq9KNRHDbpQwVBIddiz/aooa5qniiY/iqYiAro0K8SiKqS/w3quK3eAa0Qp61Sl6jS8asGrKqDYYAgMiAFQeLSEsRTHpC7VG0hAKwKgS6YnGy63o3810u6vqkm7czmim9S%2BVN68yKa3S2atgsTXKlKBtNK9AMyg1NCAo5KXYKIMLVMPmeBUwMoiW/ylqhcHmrG9olnJA3GzqoWmSkW4m%2BMxKlS6%2BFKmVNWmW7S7g97fCYc3Yd8z6JWta1WsajWrWxYHWvWpCKsYeI2jm8ms29GpozG1o7G0sy/GWC4MwUWMKwWp86S6Kp2h6wapOkaz216gqya2m3YQC1kRWwi0OhyqmiOqUbWlKGOg2%2BOk4Y2jHU2861OoKvmzO0K1QNUM2CSso26uS52hK02iumUL26uzKyk8C2EUMBRfCMbOMTAP4DgeYWgTgUWXgPwbgXgVATgNwawAaFg2sdYHgUgAgTQA%2B%2BYX2EAUWA6I%2BjgSQU%2Bl%2B0gS%2BjgXgBQEAA6Z%2BjgLQeYOAWAGARAFAQCRhegMgCgQ%2BeTTuGIYALgAEPgOgZiEBiACIf%2BiIYIeoNUTgR%2B4h5gYgNUfBCIbQbVch3gOTcyfBBgaPf%2BrAd9YAKBdhEB8%2B0gLAVmuPVYLQARvAb0BwaGH0f%2Bn3JUER3gPUSof%2BiKCIM8ahjwLAf%2B7cFgRh0gJGN2JQMfIRiKIwF%2B%2BYKgAwYABQSiL4QhRgXR/gQQEQMQdgKQGQQQT7NQf%2B3QVoAwUx0wSwawfQPAXaeACAGMEAYAd2YRlnPR/oQcYDX2LOABwkdIPhw9Q9SEK8AJqwSwLgTkXYQ9WgAbIYF4EsYIQp/BYAC%2B4nHeeAeYOwbVdIFwYyUYFoUgQIYIXoMdHIFIeiDITwZoXpvIdIKYHNPQRpyRgQToEYQZsYNoJp6Z%2BmsZnp8YemtpiZ5ZrpmNCQBplg3Z/QY%2Bv%2BiBi%2BzgXYDguIQ9OIZ9YAWXXYLBrCGyiAXAQgcGDYJ6XgcByB0gCAJANncgSgGbIwQ9Wjc8XRruV5lHPQGgI3aIQ9JCUjJ0f%2Bxx4QEFVx6QFFzx9QE50gHxp%2B70TAc0UBw%2Bo50gM%2B0RwBgiqyACOXcGfBJUDbY0C5q5m5u5h5p5l5zqd5w6DwNBpB%2B%2Bj5p%2Bsxt%2Bj%2Br%2BzgX%2B0gHRz%2Bsl/%2BwB4B0BoVk5qB2B35kARYCLJUAF1BxB6IUIVgVYZl65yGNlyQM/XgCFi8Il1oFF5x8QNxzFpQLxnFvFr4M8Rsc%2BkljgE%2BmVnFwB%2BljV4NKCQ11loOdl/OXlnVy8AEQVr51%2B0gBFrAGIK2b%2BiVqVg6cl05oB2wBV2N5J9%2B6V7%2B9YY5ilzgT54Vw5jgMwYtzNstpV%2BYXEvwSQIAA%3D)